app/Domain/Tickets/Services/Tickets.php

```diff
@@ -2059,6 +2061,157 @@ public function prepareTicketDates(&$values)
+        // using that date attribute
+        public function pollForUpdatedAccountMilestones(): array|false
+        {
+            $milestones = $this->ticketRepository->getAllBySearchCriteria(
+                ["type" => "milestone"],
+                'date'
+            );
+
+            foreach ($milestones as $key => $milestone) {
+                $milestones[$key]['id'] = $milestone['id'] . '-' . $milestone['date'];
+            }
+
+            return $milestones;
+        }
+
+        // TODO: update this to be todo and update zapier as well 
+        public function pollForNewAccountTickets(): array|false
+        {
+            return $this->ticketRepository->getAllBySearchCriteria(
+                ["excludeType" => "milestone"],
+                'date'
+            );
+        }
+
+        // Since the date attribute of todos gets updated when the todo is updated we need to poll for updated todos
+        // using that date attribute
+        public function pollForUpdatedAccountTodos(): array|false
+        {
+            $todos = $this->ticketRepository->getAllBySearchCriteria(
+                ["excludeType" => "milestone"],
+                'date'
+            );
+
+            foreach ($todos as $key => $todo) {
+                $todos[$key]['id'] = $todo['id'] . '-' . $todo['date'];
+            }
+
+            return $todos;
+        }
+
+        public function createTodoUsingZapier($values): array|bool
+        {
+            $values = array(
+                'headline' => $values['headline'] ?? "",
+                'userId' => $values['userId'] ?? "",
+                'type' => $values['type'] ?? "",
+                'description' => $values['description'] ?? "",
+                'projectId' => $values['projectId'] ?? session("currentProject"),
+                'editorId' => $values['editorId'] ?? "",
+                'date' => dtHelper()->userNow()->formatDateTimeForDb(),
+                'dateToFinish' => $values['dateToFinish'] ?? "",
+                'timeToFinish' => $values['timeToFinish'] ?? "",
+                'status' => $values['status'] ?? "",
+                'planHours' => $values['planHours'] ?? "",
+                'tags' => $values['tags'] ?? "",
+                'sprint' => $values['sprint'] ?? "",
+                'storypoints' => $values['storypoints'] ?? "",
+                'hourRemaining' => $values['hourRemaining'] ?? "",
+                'priority' => $values['priority'] ?? "",
+                'acceptanceCriteria' => $values['acceptanceCriteria'] ?? "",
+                'editFrom' => $values['editFrom'] ?? "",
+                'editTo' => $values['editTo'] ?? "",
+                'dependingTicketId' => $values['dependingTicketId'] ?? "",
+                'milestoneid' => $values['milestoneid'] ?? "",
+            );
+
+            return $this->addTicket($values);
+        }
+
+        public function updateTodoUsingZapier($values): array
```

Comment:
Can you use any of the existing update methods or does that create problems?

Reply:
If we need this method can we rename it and remove Zapier from the method name. I'd like to keep the service layer flexible and functionally useful for various use cases.

Reply:
yep lets use `updateTicket` I agree